platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -513,4 +524,6 @@ ViewportGridProvider.propTypes = {
   service: PropTypes.instanceOf(ViewportGridService).isRequired,
 };
 
-export const useViewportGrid = () => useContext(ViewportGridContext);
+// Update the useViewportGrid hook
+export const useViewportGrid = (): [AppTypes.ViewportGrid.State, ViewportGridApi] =>
```

Comment:
Thank you for adding types on this - it was not great before not having that.
In the previous version, the viewport grid set state methods weren't done immediately - that is,
viewportGrid.setState({valueKey: newValue} )
state.valueKey
didn't necessarily return the same value as newValue - eventualy that will be true, but it isn't necessarily immediate.  That should be a warning in the docs that indicate that the state may not be reflected immediately - I've seen several bugs in the code caused by that issue.  Instead, the way to handle it is to make all the changes at once OR to react to changes.

Reply:
I haven't seen anything delay wise for these Zustand stores yet. Let's open an issue when we encounter it.